src/Orleans.TelemetryConsumers.Counters/Statistics/PerfCounterEnvironmentStatistics.cs

```diff
@@ -0,0 +1,274 @@
+        {
+            get
+            {
+                return String.Format("gen0={0}, gen1={1}, gen2={2}", GC.CollectionCount(0), GC.CollectionCount(1), GC.CollectionCount(2));
+            }
+        }
+
+#if LOG_MEMORY_PERF_COUNTERS
+
+        private string GCGenSizes
+        {
+            get
+            {
+                if (genSizesPF == null) return String.Empty;
+                return String.Format("gen0={0:0.00}, gen1={1:0.00}, gen2={2:0.00}", genSizesPF[0].NextValue() / KB, genSizesPF[1].NextValue() / KB, genSizesPF[2].NextValue() / KB);
+            }
+        }
+
+#endif
+
+        public PerfCounterEnvironmentStatistics(ILoggerFactory loggerFactory)
+        {
+            this.logger = loggerFactory.CreateLogger<PerfCounterEnvironmentStatistics>();
+            try
+            {
+                Task.Run(() =>
+                {
+                    InitCpuMemoryCounters();
+                }).WaitWithThrow(INITIALIZATION_TIMEOUT);
+            }
+            catch (TimeoutException)
+            {
+                logger.Warn(ErrorCode.PerfCounterConnectError,
+                   "Timeout occurred during initialization of CPU & Memory perf counters");
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        private void InitCpuMemoryCounters()
+        {
+            try
+            {
+                cpuCounterPF = new PerformanceCounter("Processor", "% Processor Time", "_Total", true);
+                availableMemoryCounterPF = new PerformanceCounter("Memory", "Available Bytes", true);
+#if LOG_MEMORY_PERF_COUNTERS
+                string thisProcess = Process.GetCurrentProcess().ProcessName;
+                timeInGCPF = new PerformanceCounter(".NET CLR Memory", "% Time in GC", thisProcess, true);
+                genSizesPF = new PerformanceCounter[] {
+                    new PerformanceCounter(".NET CLR Memory", "Gen 0 heap size", thisProcess, true),
+                    new PerformanceCounter(".NET CLR Memory", "Gen 1 heap size", thisProcess, true),
+                    new PerformanceCounter(".NET CLR Memory", "Gen 2 heap size", thisProcess, true)
+                };
+                allocatedBytesPerSecPF = new PerformanceCounter(".NET CLR Memory", "Allocated Bytes/sec", thisProcess, true);
+                promotedMemoryFromGen1PF = new PerformanceCounter(".NET CLR Memory", "Promoted Memory from Gen 1", thisProcess, true);
+                numberOfInducedGCsPF = new PerformanceCounter(".NET CLR Memory", "# Induced GC", thisProcess, true);
+                largeObjectHeapSizePF = new PerformanceCounter(".NET CLR Memory", "Large Object Heap size", thisProcess, true);
+                promotedFinalizationMemoryFromGen0PF = new PerformanceCounter(".NET CLR Memory", "Promoted Finalization-Memory from Gen 0", thisProcess, true);
+#endif
+
+#if !(NETSTANDARD || __MonoCS__)
```

Comment:
We don't build this package for .NET Standard or Mono, do we? That's fine, it just means some code will be untested.